driver/InputFlags.cpp

```diff
@@ -124,9 +127,30 @@ void InputFlags::Parse(int argc, char* argv[])
                 Print();
 
             char short_name = FindShortName(long_name);
-
-            MapInputs[short_name].value = args[i + 1];
-            i++;
+            
+            //check for short name of in layout flags
+            if ( (short_name == 'I') || (short_name == 'O') || (short_name == 'f'))
+            {
+              //check layout flag values
+              if ((args[i + 1]) == ("NCHW")||
+                  (args[i + 1]) == ("NHWC")||
+                  (args[i + 1]) == ("NCDHW")||
+                  (args[i + 1]) == ("NDHWC"))
+              {
+                  MapInputs[short_name].value = args[i + 1];
+                  i++;
+              }
+              else
+              {
+                std::cout << "Invalid " << args[i] <<" Flag Type :: " << args[i+1] <<  std::endl;
+                MIOPEN_THROW("In/Out/Fill Layout Flags supports only  NCHW/NHWC/NCDHW/NDHWC values.");
```

Comment:
`MIOPEN_THROW` is intended for use in the library and should not be used in the external programs (like this driver)

Reply:
Ok. I will add only Error message.

Reply:
The driver should stop working and return non-zero.

Reply:
No, please.

https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1316#discussion_r767329617:
> The driver should not limit layouts. Only library knows which layouts are supported.

https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1316#pullrequestreview-829671079:
> I recommend improving the message that the library throws when invalid layout is passed to it.

That is, let the driver to pass any layout to the library (but convert to uppercase first). If something is wrong, the library (not the driver) fires MIOPEN_THROW, which contains a diagnostic message.

If that message does not suit us, then it is necessary to improve it.

The problem in #1311 is the library says "mismatched layout string, unexpect char: N" but actually passed layout was "nchw", which confuses the user. I believe you can propose and implement much better error message in the library.

Reply:
Resolved.